Client/Frontend/Toolbar+URLBar/TabLocationView.swift

```diff
@@ -49,6 +49,8 @@ class TabLocationView: UIView, FeatureFlaggable {
             updateTextWithURL()
             trackingProtectionButton.isHidden = !isValidHttpUrlProtocol
             shareButton.isHidden = !(shouldEnableShareButtonFeature && isValidHttpUrlProtocol)
+            let product = url.flatMap(ShoppingProduct.init)
+            let isShoppingCartButtonVisible = product?.isShoppingCartButtonVisible ?? false
             shoppingCartButton.isHidden = !isShoppingCartButtonVisible
```

Comment:
Hey so I got to a point where I'm personally comfortable with functional programming and I'm usually taking the short concise version. If the team feels like we prefer readability, I can change it no problem.

Reply:
Haha, I really feel you on this. I'm mostly doing stuff in LISP these days (which is basically all functional), and Haskell (where functional is strictly enforced) is the next language I'm planning to learn. I'm also very comfortable with functional and my first instinct is to do this sort of stuff. But I hold back on it, because of readability, but I don't know how the rest of the team feel about this, which is why I'm not sure what to say.

However, even in her review, @thatswinnie unpacked this as well, so maybe I'm onto something. 🤷‍♂️